segmentation/utils/signal_extraction_test.py

```diff
@@ -4,17 +4,54 @@
 from segmentation.utils import signal_extraction
 
 
+def test_positive_pixels_extraction():
+    # this function tests the functionality of positive pixels extraction
+    # where we count the number of non-zero values for each channel
+
+    # create a series of constant arrays
+    base_channel = np.zeros((10, 10))
+
+    combined_channels = np.stack((base_channel, base_channel + 1, base_channel + 2), axis=-1)
+    combined_channels = xr.DataArray(combined_channels)
+
+    coords = np.stack((np.arange(4), np.arange(4)), axis=-1)
+
+    channel_counts = signal_extraction.positive_pixels_extraction(cell_coords=coords,
+                                                                  image_data=combined_channels)
+
+    assert np.all(channel_counts == [0, 4, 4])
+
+
+def test_center_weighting_extraction():
+    # this function tests the functionality of center weighting extraction
+    # where we add a weighting scheme with more confidence toward the center
+    # before summing across each channel
+
```

Comment:
Let's use the new synthetic data here. You can modify the function to have the signal extend 1 pixel beyond the boundaries. Then show that by using this weighting, there are reduced counts of the wrong marker in the wrong cell

Reply:
Currently, the test that will be pushed is pretty clunky in that I manually force the extension of signal 1 px beyond the nucleus and membrane depending on which cells and representations are being looked at (cell 1 is currently nuclear, cell 2 is currently membrane) in the test function. I think it would be good (and looking back, what you might have actually been getting at) to have a parameter in the synthetic channel generating functions (ex. uncertainty_thickness) that will allow the user to generate signal beyond the desired nuclear or membrane boundaries.

Reply:
Also, looking back, I realize that it wasn't entirely clear if and how to test weighted_extraction at specific cell regions (for example, at the edge of cell 1 where membrane signal from cell 2 can extend into cell 1's region). This is because my assumption was that each signal_extraction function should only return the sum of the weighted signal across each channel. I originally tested that weighted_extraction sums were within an acceptable range, but that obviously fails to take into account the signal values at specific cell regions (ex. the boundary, where a cell's signal could "bleed" into other cells). Currently, I only test that center_weighted_extraction returns the expected sum values (which are hard-coded).

The easy fix off the top of my head is to return the weighted channel signal array (from which the sums are computed) so we can index into each cell (and its surroundings) as well. However, I personally think that adds to much clutter to the return value of this function.

Probably overthinking this again, but would like to clarify.

Reply:
For sure. I don't think we need to check that the value exactly matches anything in specific regions. We just want to show that the problem is reduced. 

If there was no downweighting, and membrane signal from cell 2 was bleeding into cell 1, then all of the signal within cell 1 would get counted.

With downweighting, the amount of extracted signal should be less than the total counts of membrane signal in the wrong cell. So you can check that the counts using this function are better than using the naive extraction.